spec/index.html

```diff
@@ -1252,14 +1254,189 @@ <h2>Generalized RDF Triples, Graphs, and Datasets</h2>
 
 </section>
 
+<section id="privacy">
+  <h2>Privacy Considerations</h2>
+  <p>TODO</p>
+</section>
+
+<section id="security">
+  <h2>Security Considerations</h2>
+
+  <p>The RDF Abstract Syntax is not used directly for conveying information,
+    although concrete serialization forms are specifically intended to do so.</p>
+
+  <p>Applications MAY evaluate given data to infer more assertions or to dereference IRIs,
+    invoking the security considerations of the scheme for that IRI.
+    Note in particular, the privacy issues in [[RFC3023]] section 10 for HTTP IRIs.
+    Data obtained from an inaccurate or malicious data source may lead to inaccurate or misleading conclusions,
+    as well as the dereferencing of unintended IRIs.
+    Care must be taken to align the trust in consulted resources with the sensitivity of the intended use of the data;
+    inferences of potential medical treatments would likely require
+    different trust than inferences for trip planning.</p>
+
+  <p>RDF is used to express arbitrary application data;
+    security considerations will vary by domain of use.
+    Security tools and protocols applicable to text
+    (e.g. PGP encryption, MD5 sum validation, password-protected compression)
+    may also be used on RDF documents.
+    Security/privacy protocols must be imposed which reflect the sensitivity of the embedded information.</p>
+
+  <p>RDF can express data which is presented to the user, for example, RDF Schema labels.
+    Application rendering strings retrieved from untrusted RDF documents must ensure
+    that malignant strings may not be used to mislead the reader.
```

Comment:
I don't believe this requirement can be satisfied by any applications available today, if ever.

As things stand, humans must take care to load varyingly trustworthy RDF into appropriately partitioned data stores (typically, using named graphs), and to execute queries across "all data in the store" only when recognizing that results may include malicious, erroneous, or otherwise undesirable statements (whether triples or quads).

Such undesirable results are not the responsibility of any software in the stack.

Reply:
It's taken right out of the [Turtle IANA Considerations section](https://www.w3.org/TR/turtle/#sec-mediaReg)

> Turtle can express data which is presented to the user, for example, RDF Schema labels. Application rendering strings retrieved from untrusted Turtle documents must ensure that malignant strings may not be used to mislead the reader. The security considerations in the media type registration for XML ([[RFC3023](https://www.w3.org/TR/turtle/#bib-RFC3023)] section 10) provide additional guidance around the expression of arbitrary data and markup.

I didn't want to try to tread new ground here, but ReSpec demands that specs have such a section, and this seemed to apply generally to RDF serializations. I also don't want to revisit the IANA registrations of Turtle/TriG/N-Triples/N-Quads which already have such considerations, and it would be inconsistent to say something different.

Reply:
Wow. As I understand things, that means that zero current implementations of Turtle/TriG/N-Triples/N-Quads functionality are spec compliant, because none can "ensure that malignant strings may not be used to mislead the reader" — there's just no way to do so!

Reply:
The text should say "Applications" to be clearer as it does a little further up.

The text is in a section titled "Security considerations" - things applications need to consider. It's not compliance.

Parsers aren't applications and don't render strings.
Applications choose to use parsers and data and the applications are responsible for the usage and so responsible for misleading the user.

The text is an obligation and noteworthy point on the use of the data.

Reply:
Followups to this thread should go on #11.